{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-07 22:47:48--  https://github.com/brentp/somalier/releases/download/v0.2.12/somalier\n",
      "Resolving github.com (github.com)... 140.82.112.4\n",
      "Connecting to github.com (github.com)|140.82.112.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/143888326/fdebd780-f40f-11ea-9b9d-738273996bf4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210107T224748Z&X-Amz-Expires=300&X-Amz-Signature=41478c02b44c4a9c5fb8598f586f501edbe318aba38051a6caeed19235686767&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=143888326&response-content-disposition=attachment%3B%20filename%3Dsomalier&response-content-type=application%2Foctet-stream [following]\n",
      "--2021-01-07 22:47:48--  https://github-production-release-asset-2e65be.s3.amazonaws.com/143888326/fdebd780-f40f-11ea-9b9d-738273996bf4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210107T224748Z&X-Amz-Expires=300&X-Amz-Signature=41478c02b44c4a9c5fb8598f586f501edbe318aba38051a6caeed19235686767&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=143888326&response-content-disposition=attachment%3B%20filename%3Dsomalier&response-content-type=application%2Foctet-stream\n",
      "Resolving github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.217.84.44\n",
      "Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.217.84.44|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 29851960 (28M) [application/octet-stream]\n",
      "Saving to: ‘somalier’\n",
      "\n",
      "somalier            100%[===================>]  28.47M  79.4MB/s    in 0.4s    \n",
      "\n",
      "2021-01-07 22:47:48 (79.4 MB/s) - ‘somalier’ saved [29851960/29851960]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://github.com/brentp/somalier/releases/download/v0.2.12/somalier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "!chmod 775 ./somalier "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] starting read of 0 samples\n",
      "somalier relate\n",
      "\n",
      "calculate relatedness among samples from extracted, genotype-like information\n",
      "\n",
      "Usage:\n",
      "  somalier relate [options] [extracted ...]\n",
      "\n",
      "Arguments:\n",
      "  [extracted ...]  $sample.somalier files for each sample. the first 10 are tested as a glob patterns\n",
      "\n",
      "Options:\n",
      "  -g, --groups=GROUPS        optional path  to expected groups of samples (e.g. tumor normal pairs).\n",
      "specified as comma-separated groups per line e.g.:\n",
      "    normal1,tumor1a,tumor1b\n",
      "    normal2,tumor2a\n",
      "  --sample-prefix=SAMPLE_PREFIX\n",
      "                             optional sample prefixes that can be removed to find identical samples. e.g. batch1-sampleA batch2-sampleA\n",
      "  -p, --ped=PED              optional path to a ped/fam file indicating the expected relationships among samples.\n",
      "  -d, --min-depth=MIN_DEPTH  only genotype sites with at least this depth. (default: 7)\n",
      "  --min-ab=MIN_AB            hets sites must be between min-ab and 1 - min_ab. set this to 0.2 for RNA-Seq data (default: 0.3)\n",
      "  -u, --unknown              set unknown genotypes to hom-ref. it is often preferable to use this with VCF samples that were not jointly called\n",
      "  -i, --infer                infer relationships (https://github.com/brentp/somalier/wiki/pedigree-inference)\n",
      "  -o, --output-prefix=OUTPUT_PREFIX\n",
      "                             output prefix for results. (default: somalier)\n",
      "  -h, --help                 Show this help\n",
      "\n",
      "[somalier] specify at least 1 extracted somalier file\n"
     ]
    }
   ],
   "source": [
    "! ./somalier relate\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-07 22:47:53--  https://github.com/brentp/somalier/files/4566475/sites.hg38.rna.vcf.gz\n",
      "Resolving github.com (github.com)... 140.82.113.3\n",
      "Connecting to github.com (github.com)|140.82.113.3|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://github-production-repository-file-5c1aeb.s3.amazonaws.com/143888326/4566475?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210107T224754Z&X-Amz-Expires=300&X-Amz-Signature=527b543af5b8596d292680570f7fd826496a4e9e0b1ba78768f1c2fdb077c3c4&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=143888326&response-content-disposition=attachment%3Bfilename%3Dsites.hg38.rna.vcf.gz&response-content-type=application%2Fgzip [following]\n",
      "--2021-01-07 22:47:54--  https://github-production-repository-file-5c1aeb.s3.amazonaws.com/143888326/4566475?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210107T224754Z&X-Amz-Expires=300&X-Amz-Signature=527b543af5b8596d292680570f7fd826496a4e9e0b1ba78768f1c2fdb077c3c4&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=143888326&response-content-disposition=attachment%3Bfilename%3Dsites.hg38.rna.vcf.gz&response-content-type=application%2Fgzip\n",
      "Resolving github-production-repository-file-5c1aeb.s3.amazonaws.com (github-production-repository-file-5c1aeb.s3.amazonaws.com)... 52.217.93.76\n",
      "Connecting to github-production-repository-file-5c1aeb.s3.amazonaws.com (github-production-repository-file-5c1aeb.s3.amazonaws.com)|52.217.93.76|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 266638 (260K) [application/gzip]\n",
      "Saving to: ‘sites.hg38.rna.vcf.gz’\n",
      "\n",
      "sites.hg38.rna.vcf. 100%[===================>] 260.39K  --.-KB/s    in 0.008s  \n",
      "\n",
      "2021-01-07 22:47:54 (31.0 MB/s) - ‘sites.hg38.rna.vcf.gz’ saved [266638/266638]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://github.com/brentp/somalier/files/4566475/sites.hg38.rna.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[34mPRJDB6952/\u001b[0m\n",
      "app_tarball.tar.gz\n",
      "GRCh38.no_alt_analysis_set.fa.fai\n",
      "GRCh38.no_alt_analysis_set.fa.gz\n",
      "\u001b[1m\u001b[32mmutect\u001b[0m\n",
      "muTect-1.1.4.jar\n",
      "\u001b[1m\u001b[32mPLINK GWAS Workflow\u001b[0m\n",
      "\u001b[1m\u001b[32mplink-vcf\u001b[0m\n",
      "somalier_test.ipynb\n",
      "\u001b[1m\u001b[32mUntitled Workflow - 1/7/2021 1:10 PM\u001b[0m\n",
      "\u001b[1m\u001b[32mUntitled Workflow - Jan 7th 2021 9:27am\u001b[0m\n",
      "Untitled.ipynb\n",
      "\u001b[1m\u001b[32mvcf2ld\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!dx ls\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DRR131915.cancer.vcf\n",
      "test_reads.cancer.vcf\n",
      "DRR131921.cancer.vcf\n",
      "DRR131916.cancer.vcf\n",
      "DRR131926.cancer.vcf\n",
      "DRR131939.cancer.vcf\n",
      "DRR131927.cancer.vcf\n",
      "DRR131951.cancer.vcf\n",
      "DRR131972.cancer.vcf\n",
      "DRR131962.cancer.vcf\n"
     ]
    }
   ],
   "source": [
    "! dx ls PRJDB6952/*.vcf | head\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Untitled.ipynb\t sites.hg38.rna.vcf.gz\tsomalier_test.ipynb\n",
      "data_cancer.vcf  somalier\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘data_cancer.vcf’: File exists\n"
     ]
    }
   ],
   "source": [
    "!mkdir data_cancer.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!dx download -f PRJDB6952/*.cancer.vcf -o data_cancer.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir data_HC.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!dx download -f PRJDB6952/*.HC*.vcf.gz -o data_HC.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-07 22:54:12--  https://data.broadinstitute.org/Trinity/TMP/ctat.hg38.fasta\n",
      "Resolving data.broadinstitute.org (data.broadinstitute.org)... 69.173.92.29\n",
      "Connecting to data.broadinstitute.org (data.broadinstitute.org)|69.173.92.29|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3151417447 (2.9G)\n",
      "Saving to: ‘ctat.hg38.fasta’\n",
      "\n",
      "ctat.hg38.fasta     100%[===================>]   2.93G  11.6MB/s    in 5m 19s  \n",
      "\n",
      "2021-01-07 22:59:31 (9.43 MB/s) - ‘ctat.hg38.fasta’ saved [3151417447/3151417447]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://data.broadinstitute.org/Trinity/TMP/ctat.hg38.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Untitled.ipynb\t data_HC.vcf\t  sites.hg38.rna.vcf.gz  somalier_test.ipynb\n",
      "ctat.hg38.fasta  data_cancer.vcf  somalier\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[E::idx_find_and_load] Could not retrieve index file for 'data_cancer.vcf/DRR131561.cancer.vcf'\n",
      "hts-nim/vcf no index found for data_cancer.vcf/DRR131561.cancer.vcf\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d extracted/ --sites sites.hg38.rna.vcf.gz -f ctat.hg38.fasta data_cancer.vcf/DRR131561.cancer.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp data_cancer.vcf/DRR131561.cancer.vcf ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-07 23:22:56--  https://github.com/samtools/bcftools/releases/download/1.11/bcftools-1.11.tar.bz2\n",
      "Resolving github.com (github.com)... 140.82.113.4\n",
      "Connecting to github.com (github.com)|140.82.113.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/11368595/ea365000-fcf0-11ea-9121-5d13fc9c211d?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210107T232256Z&X-Amz-Expires=300&X-Amz-Signature=0fb609381a7eb2e085df0a9caa506189602f1e59b80fb988d0a414a5578d1764&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=11368595&response-content-disposition=attachment%3B%20filename%3Dbcftools-1.11.tar.bz2&response-content-type=application%2Foctet-stream [following]\n",
      "--2021-01-07 23:22:56--  https://github-production-release-asset-2e65be.s3.amazonaws.com/11368595/ea365000-fcf0-11ea-9121-5d13fc9c211d?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210107T232256Z&X-Amz-Expires=300&X-Amz-Signature=0fb609381a7eb2e085df0a9caa506189602f1e59b80fb988d0a414a5578d1764&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=11368595&response-content-disposition=attachment%3B%20filename%3Dbcftools-1.11.tar.bz2&response-content-type=application%2Foctet-stream\n",
      "Resolving github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.216.236.35\n",
      "Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.216.236.35|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3827193 (3.6M) [application/octet-stream]\n",
      "Saving to: ‘bcftools-1.11.tar.bz2’\n",
      "\n",
      "bcftools-1.11.tar.b 100%[===================>]   3.65M  --.-KB/s    in 0.06s   \n",
      "\n",
      "2021-01-07 23:22:56 (58.9 MB/s) - ‘bcftools-1.11.tar.bz2’ saved [3827193/3827193]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://github.com/samtools/bcftools/releases/download/1.11/bcftools-1.11.tar.bz2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!tar xvjf bcftools-1.11.tar.bz2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir bcftools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!bcftools-1.11/configure --prefix /opt/notebooks/bcftools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: bcftools/make: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!cd bcftools-1.11/\n",
    "!make\n",
    "!make install\n",
    "!cd "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools view DRR131561.cancer.vcf -Oz -o DRR131561.cancer.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools index DRR131561.cancer.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] found 136 sites\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d extracted/ --sites sites.hg38.rna.vcf.gz -f ctat.hg38.fasta ./DRR131561.cancer.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] starting read of 1 samples\n",
      "[somalier] time to read files and get per-sample stats for 1 samples: 0.00\n",
      "[somalier] time to get expected relatedness from pedigree graph: 0.00\n",
      "[somalier] time to calculate all vs all relatedness for all 0 combinations: 0.00\n",
      "[somalier] wrote interactive HTML output for 0 pairs to: somalier.html\n",
      "[somalier] wrote groups to: somalier.groups.tsv (look at this for cancer samples)\n",
      "[somalier] wrote samples to: somalier.samples.tsv\n",
      "[somalier] wrote pair-wise relatedness metrics to: somalier.pairs.tsv\n"
     ]
    }
   ],
   "source": [
    "!./somalier relate extracted/*.somalier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-07 23:45:37--  https://dl.dnanex.us/F/D/2Xpf81q00x98vK0Bzq0j7Bp2Pk1jbXZfqzP2KK1V/compiled.cancer.vcf.gz\n",
      "Resolving dl.dnanex.us (dl.dnanex.us)... 34.233.40.46, 34.232.176.39\n",
      "Connecting to dl.dnanex.us (dl.dnanex.us)|34.233.40.46|:443... connected.\n",
      "HTTP request sent, awaiting response... 401 Unauthorized\n",
      "\n",
      "Username/Password Authentication Failed.\n"
     ]
    }
   ],
   "source": [
    "!wget https://dl.dnanex.us/F/D/2Xpf81q00x98vK0Bzq0j7Bp2Pk1jbXZfqzP2KK1V/compiled.cancer.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2K[===========================================================>] Completed 842,373 of 842,373 bytes (100%) /opt/notebooks/compiled.cancer.vcf.gzz\n"
     ]
    }
   ],
   "source": [
    "!dx download compiled.cancer.vcf.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2K[===========================================================>] Completed 2,946,390 of 2,946,390 bytes (100%) /opt/notebooks/HC_hard_cutoffs.vcf.gzz\n"
     ]
    }
   ],
   "source": [
    "!dx download HC_hard_cutoffs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip HC_hard_cutoffs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools view HC_hard_cutoffs.vcf -Oz -o HC_hard_cutoffs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools index HC_hard_cutoffs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools view compiled.cancer.vcf -Oz -o compiled.cancer.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools index compiled.cancer.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] found 1734 sites\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d extracted/ --sites sites.hg38.vcf.gz -f ctat.hg38.fasta compiled.cancer.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] starting read of 122 samples\n",
      "[somalier] time to read files and get per-sample stats for 122 samples: 0.07\n",
      "[somalier] time to get expected relatedness from pedigree graph: 0.00\n",
      "[somalier] time to calculate all vs all relatedness for all 7381 combinations: 0.06\n",
      "[somalier] wrote interactive HTML output for 7381 pairs to: somalier.html\n",
      "[somalier] joining families DRR131893_fastq and DRR131561_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131893_fastq and DRR131576_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131893_fastq and DRR132407_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131893_fastq and DRR131905_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131893_fastq and DRR132413_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131893_fastq and DRR131570_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131893_fastq and DRR132368_fastq because of relatedness > 0.2\n",
      "[somalier] wrote groups to: somalier.groups.tsv (look at this for cancer samples)\n",
      "[somalier] wrote samples to: somalier.samples.tsv\n",
      "[somalier] wrote pair-wise relatedness metrics to: somalier.pairs.tsv\n"
     ]
    }
   ],
   "source": [
    "!./somalier relate extracted/*.somalier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] subset from 17384 to 0 high call-rate sites (removed 100.00%)\n",
      "p_shapeshifting.nim(106) broadcast2Impl\n",
      "Error: unhandled exception: Broadcasting error: non-singleton dimensions must be the same in both tensors. Tensors' shapes were: [2504, 0] and [] [ValueError]\n"
     ]
    }
   ],
   "source": [
    "!./somalier ancestry --labels ancestry-labels-1kg.tsv 1kg-somalier/*.somalier ++ extracted/*.somalier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir HC_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] found 3535 sites\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d HC_extracted/ --sites sites.hg38.rna.vcf.gz -f ctat.hg38.fasta ./HC_hard_cutoffs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] starting read of 122 samples\n",
      "[somalier] time to read files and get per-sample stats for 122 samples: 0.07\n",
      "[somalier] time to get expected relatedness from pedigree graph: 0.00\n",
      "[somalier] time to calculate all vs all relatedness for all 7381 combinations: 0.07\n",
      "[somalier] wrote interactive HTML output for 7381 pairs to: somalier.html\n",
      "[somalier] joining families DRR132035_fastq and DRR131901_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132035_fastq and DRR131760_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132035_fastq and DRR132391_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132490_fastq and DRR132035_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132490_fastq and DRR131561_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132490_fastq and DRR132479_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132490_fastq and DRR132368_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131652_fastq and DRR132490_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131652_fastq and DRR131890_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR131633_fastq and DRR131652_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132413_fastq and DRR131633_fastq because of relatedness > 0.2\n",
      "[somalier] joining families DRR132413_fastq and DRR131728_fastq because of relatedness > 0.2\n",
      "[somalier] wrote groups to: somalier.groups.tsv (look at this for cancer samples)\n",
      "[somalier] wrote samples to: somalier.samples.tsv\n",
      "[somalier] wrote pair-wise relatedness metrics to: somalier.pairs.tsv\n"
     ]
    }
   ],
   "source": [
    "!./somalier relate HC_extracted_2/*.somalier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-08 00:22:42--  https://raw.githubusercontent.com/brentp/somalier/master/scripts/ancestry-labels-1kg.tsv\n",
      "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.200.133\n",
      "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.200.133|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 56028 (55K) [text/plain]\n",
      "Saving to: ‘ancestry-labels-1kg.tsv’\n",
      "\n",
      "ancestry-labels-1kg 100%[===================>]  54.71K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2021-01-08 00:22:42 (82.6 MB/s) - ‘ancestry-labels-1kg.tsv’ saved [56028/56028]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://raw.githubusercontent.com/brentp/somalier/master/scripts/ancestry-labels-1kg.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-08 00:30:33--  https://zenodo.org/record/3479773/files/1kg.somalier.tar.gz\n",
      "Resolving zenodo.org (zenodo.org)... 137.138.76.77\n",
      "Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 82856769 (79M) [application/octet-stream]\n",
      "Saving to: ‘1kg.somalier.tar.gz’\n",
      "\n",
      "1kg.somalier.tar.gz 100%[===================>]  79.02M  18.7MB/s    in 4.7s    \n",
      "\n",
      "2021-01-08 00:30:39 (16.7 MB/s) - ‘1kg.somalier.tar.gz’ saved [82856769/82856769]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://zenodo.org/record/3479773/files/1kg.somalier.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!tar -xvf 1kg.somalier.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "fatal.nim(49)            sysFatal\n",
      "Error: unhandled exception: ancestry.nim(138, 14) `cnt.sites.len == train_mat[0].len` bad number of sites 16469 in extracted/DRR131561_fastq.somalier [AssertionError]\n"
     ]
    }
   ],
   "source": [
    "!./somalier ancestry --labels ancestry-labels-1kg.tsv 1kg-somalier/*.somalier ++ extracted/*.somalier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-08 01:02:10--  https://github.com/brentp/somalier/files/3412456/sites.hg38.vcf.gz\n",
      "Resolving github.com (github.com)... 140.82.112.4\n",
      "Connecting to github.com (github.com)|140.82.112.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://github-production-repository-file-5c1aeb.s3.amazonaws.com/143888326/3412456?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210108%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210108T010210Z&X-Amz-Expires=300&X-Amz-Signature=8611638c60174fd9c6526dd977fafac1c11cfacac0b6e576692d1d597783b832&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=143888326&response-content-disposition=attachment%3Bfilename%3Dsites.hg38.vcf.gz&response-content-type=application%2Fgzip [following]\n",
      "--2021-01-08 01:02:10--  https://github-production-repository-file-5c1aeb.s3.amazonaws.com/143888326/3412456?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210108%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210108T010210Z&X-Amz-Expires=300&X-Amz-Signature=8611638c60174fd9c6526dd977fafac1c11cfacac0b6e576692d1d597783b832&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=143888326&response-content-disposition=attachment%3Bfilename%3Dsites.hg38.vcf.gz&response-content-type=application%2Fgzip\n",
      "Resolving github-production-repository-file-5c1aeb.s3.amazonaws.com (github-production-repository-file-5c1aeb.s3.amazonaws.com)... 52.217.79.236\n",
      "Connecting to github-production-repository-file-5c1aeb.s3.amazonaws.com (github-production-repository-file-5c1aeb.s3.amazonaws.com)|52.217.79.236|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 265818 (260K) [application/gzip]\n",
      "Saving to: ‘sites.hg38.vcf.gz’\n",
      "\n",
      "sites.hg38.vcf.gz   100%[===================>] 259.59K  --.-KB/s    in 0.003s  \n",
      "\n",
      "2021-01-08 01:02:10 (83.3 MB/s) - ‘sites.hg38.vcf.gz’ saved [265818/265818]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://github.com/brentp/somalier/files/3412456/sites.hg38.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] found 2059 sites\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d HC_extracted_2/ --sites sites.hg38.vcf.gz -f ctat.hg38.fasta ./HC_hard_cutoffs.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools index data_HC.vcf/DRR131564.variants.HC_init.wAnnot.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] found 140 sites\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d HC_extracted_3/ --sites sites.hg38.vcf.gz -f ctat.hg38.fasta data_HC.vcf/DRR131564.variants.HC_init.wAnnot.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools index  data_HC.vcf/DRR131647.variants.HC_init.wAnnot.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "!for i in data_HC.vcf/*wAnnot.vcf.gz; do bcftools index -f $i; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] found 116 sites\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d HC_extracted_3/ --sites sites.hg38.vcf.gz -f ctat.hg38.fasta data_HC.vcf/DRR131647.variants.HC_init.wAnnot.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "somalier extract\n",
      "\n",
      "extract genotype-like information for a single-sample at selected sites\n",
      "\n",
      "Usage:\n",
      "  somalier extract [options] sample_file\n",
      "\n",
      "Arguments:\n",
      "  sample_file      single-sample CRAM/BAM/GVCF file or multi/single-sample VCF from which to extract\n",
      "\n",
      "Options:\n",
      "  -s, --sites=SITES          sites vcf file of variants to extract\n",
      "  -f, --fasta=FASTA          path to reference fasta file\n",
      "  -d, --out-dir=OUT_DIR      path to output directory (default: .)\n",
      "  --sample-prefix=SAMPLE_PREFIX\n",
      "                             prefix for the sample name stored inside the digest\n",
      "  -h, --help                 Show this help\n"
     ]
    }
   ],
   "source": [
    "!./somalier extract -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "files.nim(30)            file_type\n",
      "Error: unhandled exception: unable to open file:for [OSError]\n",
      "somalier version: 0.2.12\n",
      "files.nim(30)            file_type\n",
      "Error: unhandled exception: unable to open file:i [OSError]\n",
      "somalier version: 0.2.12\n",
      "files.nim(30)            file_type\n",
      "Error: unhandled exception: unable to open file:in [OSError]\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 109 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 140 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 192 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 184 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 167 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 180 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 205 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 247 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 199 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 193 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 257 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 233 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 231 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 123 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 60 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 116 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 136 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 181 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 101 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 178 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 149 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 108 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 119 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 179 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 175 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 124 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 124 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 71 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 101 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 183 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 132 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 94 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 106 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 191 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 147 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 58 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 238 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 139 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 123 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 274 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 253 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 80 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 289 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 181 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 236 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 222 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 204 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 377 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 178 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 174 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 243 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 183 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 258 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 189 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 70 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 120 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 166 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 336 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 132 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 152 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 251 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 175 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 150 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 103 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 120 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 140 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 337 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 324 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 156 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 70 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 191 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 204 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 159 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 256 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 209 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 235 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 94 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 90 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 271 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 139 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 250 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 284 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 96 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 172 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 252 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 268 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 198 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 92 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 186 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 259 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 265 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 189 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 182 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 157 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 189 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 302 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 316 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 155 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 281 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 99 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 258 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 242 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 170 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 218 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 188 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 290 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 269 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 179 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 100 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 246 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 138 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 144 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 152 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 234 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 119 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 143 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 96 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 123 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 357 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 327 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 186 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 35 sites\n",
      "somalier version: 0.2.12\n",
      "[somalier] found 0 sites\n"
     ]
    }
   ],
   "source": [
    "!for i in for i in data_HC.vcf/*wAnnot.vcf.gz; do ./somalier extract -d HC_extracted_3/ --sites sites.hg38.vcf.gz -f ctat.hg38.fasta $i; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] subset from 17384 to 2 high call-rate sites (removed 99.99%)\n",
      "[somalier] time for dimensionality reduction to shape [2504, 5]: 0.07 seconds\n",
      "[somalier] Epoch:0. loss: 1.46617. accuracy on unseen data: 0.366.  total-time: 0.00\n",
      "[somalier] Epoch:500. loss: 1.23110. accuracy on unseen data: 0.426.  total-time: 1.57\n",
      "[somalier] Epoch:1000. loss: 1.21974. accuracy on unseen data: 0.436.  total-time: 3.12\n",
      "[somalier] Epoch:1500. loss: 1.20833. accuracy on unseen data: 0.436.  total-time: 4.69\n",
      "[somalier] Epoch:2000. loss: 1.19582. accuracy on unseen data: 0.465.  total-time: 6.24\n",
      "[somalier] Epoch:2500. loss: 1.21622. accuracy on unseen data: 0.446.  total-time: 7.80\n",
      "[somalier] Epoch:3000. loss: 1.21877. accuracy on unseen data: 0.485.  total-time: 9.33\n",
      "[somalier] Epoch:3500. loss: 1.21507. accuracy on unseen data: 0.465.  total-time: 10.92\n",
      "[somalier] Epoch:4000. loss: 1.22530. accuracy on unseen data: 0.446.  total-time: 12.48\n",
      "[somalier] Epoch:4500. loss: 1.21326. accuracy on unseen data: 0.436.  total-time: 14.01\n",
      "[somalier] Epoch:5000. loss: 1.18483. accuracy on unseen data: 0.465.  total-time: 15.57\n",
      "[somalier] Epoch:5500. loss: 1.19570. accuracy on unseen data: 0.436.  total-time: 17.14\n",
      "[somalier] Epoch:6000. loss: 1.18768. accuracy on unseen data: 0.465.  total-time: 18.71\n",
      "[somalier] Epoch:6500. loss: 1.20717. accuracy on unseen data: 0.475.  total-time: 20.26\n",
      "[somalier] Epoch:7000. loss: 1.20635. accuracy on unseen data: 0.436.  total-time: 21.83\n",
      "[somalier] Epoch:7500. loss: 1.22073. accuracy on unseen data: 0.436.  total-time: 23.38\n",
      "[somalier] Epoch:8000. loss: 1.21411. accuracy on unseen data: 0.436.  total-time: 24.91\n",
      "[somalier] Epoch:8500. loss: 1.21664. accuracy on unseen data: 0.455.  total-time: 26.46\n",
      "[somalier] Epoch:9000. loss: 1.20550. accuracy on unseen data: 0.426.  total-time: 28.01\n",
      "[somalier] Epoch:9500. loss: 1.21318. accuracy on unseen data: 0.446.  total-time: 29.62\n",
      "[somalier] Epoch:10000. loss: 1.19358. accuracy on unseen data: 0.455.  total-time: 31.19\n",
      "[somalier] reduced query set to: [123, 5]\n",
      "[somalier] wrote text file to somalier-ancestry.somalier-ancestry.tsv\n",
      "[somalier] wrote html file to somalier-ancestry.somalier-ancestry.html\n"
     ]
    }
   ],
   "source": [
    "!./somalier ancestry --labels ancestry-labels-1kg.tsv 1kg-somalier/*.somalier ++ HC_extracted_3/*.somalier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "somalier pca\n",
      "\n",
      "dimensionality reduction\n",
      "\n",
      "Usage:\n",
      "  somalier pca [options] [extracted ...]\n",
      "\n",
      "Arguments:\n",
      "  [extracted ...]  $sample.somalier files for each sample. place labelled samples first followed by '++' then *.somalier for query samples\n",
      "\n",
      "Options:\n",
      "  --labels=LABELS            file with ancestry labels\n",
      "  -o, --output-prefix=OUTPUT_PREFIX\n",
      "                             prefix for output files (default: somalier-ancestry)\n",
      "  --n-pcs=N_PCS              number of principal components to use in the reduced dataset (default: 5)\n",
      "  --nn-hidden-size=NN_HIDDEN_SIZE\n",
      "                             shape of hidden layer in neural network (default: 16)\n",
      "  --nn-batch-size=NN_BATCH_SIZE\n",
      "                             batch size fo training neural network (default: 32)\n",
      "  --nn-test-samples=NN_TEST_SAMPLES\n",
      "                             number of labeled samples to test for NN convergence (default: 101)\n",
      "  -h, --help                 Show this help\n"
     ]
    }
   ],
   "source": [
    "!./somalier ancestry \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[===========================================================>] Uploaded 560,396 of 560,396 bytes (100%) somalier.html\n",
      "ID                  file-Fzqp6y8005ZgvkbjJKQjy28k\n",
      "Class               file\n",
      "Project             project-FzpxXqj005ZXPV7b3g4gJYB0\n",
      "Folder              /\n",
      "Name                somalier.html\n",
      "State               \u001b[33mclosing\u001b[0m\n",
      "Visibility          visible\n",
      "Types               -\n",
      "Properties          -\n",
      "Tags                -\n",
      "Outgoing links      -\n",
      "Created             Fri Jan  8 00:12:41 2021\n",
      "Created by          arhodes\n",
      " via the job        job-Fzqjv5Q005ZvjqYq5v879x5p\n",
      "Last modified       Fri Jan  8 00:12:41 2021\n",
      "Media type          \n",
      "archivalState       \"live\"\n",
      "cloudAccount        \"cloudaccount-dnanexus\"\n"
     ]
    }
   ],
   "source": [
    "!dx upload somalier.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Trying a new data set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-08 05:05:15--  https://download.su92l.arvadosapi.com/c=9001fae4044d8baadd83d4035a019d9e-495/_/AH8LD7MW6WE.vcf.gz\n",
      "Resolving download.su92l.arvadosapi.com (download.su92l.arvadosapi.com)... 52.165.170.22\n",
      "Connecting to download.su92l.arvadosapi.com (download.su92l.arvadosapi.com)|52.165.170.22|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 695830191 (664M) [application/gzip]\n",
      "Saving to: ‘AH8LD7MW6WE.vcf.gz’\n",
      "\n",
      "AH8LD7MW6WE.vcf.gz  100%[===================>] 663.59M  30.7MB/s    in 28s     \n",
      "\n",
      "2021-01-08 05:05:43 (23.8 MB/s) - ‘AH8LD7MW6WE.vcf.gz’ saved [695830191/695830191]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://download.su92l.arvadosapi.com/c=9001fae4044d8baadd83d4035a019d9e-495/_/AH8LD7MW6WE.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools index AH8LD7MW6WE.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "somalier version: 0.2.12\n",
      "[somalier] found 77 sites\n"
     ]
    }
   ],
   "source": [
    "! ./somalier extract -d PGP_test/ --sites sites.hg38.vcf.gz -f ctat.hg38.fasta AH8LD7MW6WE.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[===========================================================>] Uploaded 14,085 of 14,085 bytes (100%) somalier.samples.tsv\n",
      "ID                  file-FzqzFzj005ZfkpFq7Q41bY8B\n",
      "Class               file\n",
      "Project             project-FzpxXqj005ZXPV7b3g4gJYB0\n",
      "Folder              /\n",
      "Name                somalier.samples.tsv\n",
      "State               \u001b[33mclosing\u001b[0m\n",
      "Visibility          visible\n",
      "Types               -\n",
      "Properties          -\n",
      "Tags                -\n",
      "Outgoing links      -\n",
      "Created             Fri Jan  8 06:04:47 2021\n",
      "Created by          arhodes\n",
      " via the job        job-Fzqjv5Q005ZvjqYq5v879x5p\n",
      "Last modified       Fri Jan  8 06:04:47 2021\n",
      "Media type          \n",
      "archivalState       \"live\"\n",
      "cloudAccount        \"cloudaccount-dnanexus\"\n"
     ]
    }
   ],
   "source": [
    "!dx upload somalier.samples.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[===========================================================>] Uploaded 542,502 of 542,502 bytes (100%) somalier.pairs.tsv\n",
      "ID                  file-FzqzFq0005ZpGqgBGJfGjJKq\n",
      "Class               file\n",
      "Project             project-FzpxXqj005ZXPV7b3g4gJYB0\n",
      "Folder              /\n",
      "Name                somalier.pairs.tsv\n",
      "State               \u001b[33mclosing\u001b[0m\n",
      "Visibility          visible\n",
      "Types               -\n",
      "Properties          -\n",
      "Tags                -\n",
      "Outgoing links      -\n",
      "Created             Fri Jan  8 06:04:28 2021\n",
      "Created by          arhodes\n",
      " via the job        job-Fzqjv5Q005ZvjqYq5v879x5p\n",
      "Last modified       Fri Jan  8 06:04:28 2021\n",
      "Media type          \n",
      "archivalState       \"live\"\n",
      "cloudAccount        \"cloudaccount-dnanexus\"\n"
     ]
    }
   ],
   "source": [
    "!dx upload somalier.pairs.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[===========================================================>] Uploaded 128,328 of 128,328 bytes (100%) somalier.groups.tsv\n",
      "ID                  file-FzqzFfj005ZQG3py3kgx3b3f\n",
      "Class               file\n",
      "Project             project-FzpxXqj005ZXPV7b3g4gJYB0\n",
      "Folder              /\n",
      "Name                somalier.groups.tsv\n",
      "State               \u001b[33mclosing\u001b[0m\n",
      "Visibility          visible\n",
      "Types               -\n",
      "Properties          -\n",
      "Tags                -\n",
      "Outgoing links      -\n",
      "Created             Fri Jan  8 06:04:11 2021\n",
      "Created by          arhodes\n",
      " via the job        job-Fzqjv5Q005ZvjqYq5v879x5p\n",
      "Last modified       Fri Jan  8 06:04:11 2021\n",
      "Media type          \n",
      "archivalState       \"live\"\n",
      "cloudAccount        \"cloudaccount-dnanexus\"\n"
     ]
    }
   ],
   "source": [
    "!dx upload somalier.groups.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[===========================================================>] Uploaded 234,625 of 234,625 bytes (100%) somalier-ancestry.somalier-ancestry.tsv\n",
      "ID                  file-FzqzFY8005Zb8FF16f5q4928\n",
      "Class               file\n",
      "Project             project-FzpxXqj005ZXPV7b3g4gJYB0\n",
      "Folder              /\n",
      "Name                somalier-ancestry.somalier-ancestry.tsv\n",
      "State               \u001b[33mclosing\u001b[0m\n",
      "Visibility          visible\n",
      "Types               -\n",
      "Properties          -\n",
      "Tags                -\n",
      "Outgoing links      -\n",
      "Created             Fri Jan  8 06:03:57 2021\n",
      "Created by          arhodes\n",
      " via the job        job-Fzqjv5Q005ZvjqYq5v879x5p\n",
      "Last modified       Fri Jan  8 06:03:57 2021\n",
      "Media type          \n",
      "archivalState       \"live\"\n",
      "cloudAccount        \"cloudaccount-dnanexus\"\n"
     ]
    }
   ],
   "source": [
    "!dx upload somalier-ancestry.somalier-ancestry.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[===========================================================>] Uploaded 401,308 of 401,308 bytes (100%) somalier-ancestry.somalier-ancestry.html\n",
      "ID                  file-Fzqz3xQ005ZQxg363p98Q3J6\n",
      "Class               file\n",
      "Project             project-FzpxXqj005ZXPV7b3g4gJYB0\n",
      "Folder              /\n",
      "Name                somalier-ancestry.somalier-ancestry.html\n",
      "State               \u001b[33mclosing\u001b[0m\n",
      "Visibility          visible\n",
      "Types               -\n",
      "Properties          -\n",
      "Tags                -\n",
      "Outgoing links      -\n",
      "Created             Fri Jan  8 05:47:34 2021\n",
      "Created by          arhodes\n",
      " via the job        job-Fzqjv5Q005ZvjqYq5v879x5p\n",
      "Last modified       Fri Jan  8 05:47:34 2021\n",
      "Media type          \n",
      "archivalState       \"live\"\n",
      "cloudAccount        \"cloudaccount-dnanexus\"\n"
     ]
    }
   ],
   "source": [
    "!dx upload somalier-ancestry.somalier-ancestry.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[===========================================================>] Uploaded 56,685 of 56,685 bytes (100%) somalier_test.ipynb\n",
      "ID                  file-FzqzG30005ZQvKZxKbjq02Zp\n",
      "Class               file\n",
      "Project             project-FzpxXqj005ZXPV7b3g4gJYB0\n",
      "Folder              /\n",
      "Name                somalier_test.ipynb\n",
      "State               \u001b[33mclosing\u001b[0m\n",
      "Visibility          visible\n",
      "Types               -\n",
      "Properties          -\n",
      "Tags                -\n",
      "Outgoing links      -\n",
      "Created             Fri Jan  8 06:05:00 2021\n",
      "Created by          arhodes\n",
      " via the job        job-Fzqjv5Q005ZvjqYq5v879x5p\n",
      "Last modified       Fri Jan  8 06:05:00 2021\n",
      "Media type          \n",
      "archivalState       \"live\"\n",
      "cloudAccount        \"cloudaccount-dnanexus\"\n"
     ]
    }
   ],
   "source": [
    "!dx upload somalier_test.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
